router/core/transport.go

```diff
@@ -322,20 +329,24 @@ func NewTransport(opts *TransportOptions) *TransportFactory {
 		preHandlers:                   opts.PreHandlers,
 		postHandlers:                  opts.PostHandlers,
 		retryOptions:                  opts.RetryOptions,
-		requestTimeout:                opts.RequestTimeout,
+		subgraphTransportOptions:      opts.SubgraphTransportOptions,
 		localhostFallbackInsideDocker: opts.LocalhostFallbackInsideDocker,
 		metricStore:                   opts.MetricStore,
 		logger:                        opts.Logger,
 		tracerProvider:                opts.TracerProvider,
+		proxy:                         opts.Proxy,
 	}
 }
 
-func (t TransportFactory) RoundTripper(enableSingleFlight bool, transport http.RoundTripper) http.RoundTripper {
+func (t TransportFactory) RoundTripper(enableSingleFlight bool, baseTransport http.RoundTripper) http.RoundTripper {
+	baseTransport = NewTimeoutTransport(t.subgraphTransportOptions, baseTransport, t.logger, t.proxy)
```

Comment:
NewTimeoutTransport should only be applied when someone defined custom subgraph timeouts. The transport has overhead. Overhead we can remove in most of the cases.

Reply:
Good catch! removed it